portfolio/src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -45,23 +56,127 @@ public void tearDown() {
     helper.tearDown();
   }
 
+  @Test 
+  public void testPOSTSingleComment() throws Exception {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    HttpServletRequest postRequest = mock(HttpServletRequest.class);
+    HttpServletResponse postResponse = mock(HttpServletResponse.class);
+
+    Assert.assertEquals(0, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+
+    when(postRequest.getParameter("text-input")).thenReturn("Comment1");
+    new DataServlet().doPost(postRequest, postResponse);
+    
+    Assert.assertEquals(1, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+  }
+
+  @Test 
+  public void testPOSTMultipleComments() throws Exception {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    HttpServletRequest postRequest = mock(HttpServletRequest.class);
+    HttpServletResponse postResponse = mock(HttpServletResponse.class);
+
+    Assert.assertEquals(0, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+
+    when(postRequest.getParameter("text-input")).thenReturn("Comment1");
+    new DataServlet().doPost(postRequest, postResponse);
+    when(postRequest.getParameter("text-input")).thenReturn("Comment2");
+    new DataServlet().doPost(postRequest, postResponse);
+    when(postRequest.getParameter("text-input")).thenReturn("Comment3");
+    new DataServlet().doPost(postRequest, postResponse);
+    
+    Assert.assertEquals(3, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+  }
+
+  @Test 
+  public void testPOSTNull() throws Exception {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+    HttpServletRequest postRequest = mock(HttpServletRequest.class);
+    HttpServletResponse postResponse = mock(HttpServletResponse.class);
+
+    Assert.assertEquals(0, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+
+    when(postRequest.getParameter("text-input")).thenReturn(null);
+    new DataServlet().doPost(postRequest, postResponse);
+    
+    Assert.assertEquals(0, ds.prepare(new Query(COMMENT_TABLE_NAME)).countEntities(withLimit(10)));
+  }
+
+  @Test
+  public void testGETSingleComment() throws Exception {
+    HttpServletRequest postRequest = mock(HttpServletRequest.class);
+    HttpServletResponse postResponse = mock(HttpServletResponse.class);
+    HttpServletRequest getRequest = mock(HttpServletRequest.class);
+    HttpServletResponse getResponse = mock(HttpServletResponse.class);
+
+    when(postRequest.getParameter("text-input")).thenReturn("Test Comment #1");
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter printWriter = new PrintWriter(stringWriter);
+    when(getResponse.getWriter()).thenReturn(printWriter);
+
+    new DataServlet().doPost(postRequest, postResponse);
```

Comment:
nit - instead of relying on the post method to populate this data can this test just use the Datastore directly?

Reply:
sounds good! will do!